tests/src/test/java/org/sonarsource/tests/SamlTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * SAML 2.0 Authentication for SonarQube
+ * Copyright (C) 2018-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.tests;
+
+import com.codeborne.selenide.Condition;
+import com.sonar.orchestrator.Orchestrator;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.sonarqube.ws.WsUsers;
+import org.sonarqube.ws.WsUsers.GroupsWsResponse.Group;
+import org.sonarqube.ws.WsUsers.SearchWsResponse.User;
+import org.sonarqube.ws.client.user.GroupsRequest;
+import org.sonarqube.ws.client.user.SearchRequest;
+import org.sonarqube.ws.client.user.UsersService;
+import org.sonarqube.ws.client.usergroup.CreateWsRequest;
+import org.sonarqube.ws.client.usergroup.DeleteWsRequest;
+import org.sonarsource.util.KeycloakServices;
+
+import static com.codeborne.selenide.Selenide.$;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarsource.util.ItUtils.SAML_CLIENT_ID;
+import static org.sonarsource.util.ItUtils.SAML_REAML;
+import static org.sonarsource.util.ItUtils.SAML_SERVER_URL;
+import static org.sonarsource.util.ItUtils.newAdminWsClient;
+import static org.sonarsource.util.ItUtils.openBrowser;
+import static org.sonarsource.util.ItUtils.setGlobalSetting;
+
+public class SamlTest {
+
+  @ClassRule
+  public static Orchestrator orchestrator = SamlTestSuite.ORCHESTRATOR;
+
+  @Rule
+  public KeycloakServices keycloakServices = new KeycloakServices();
+
+  @After
+  public void deleteUsers() {
+    UsersService users = newAdminWsClient(orchestrator).users();
+    users.search(SearchRequest.builder().build()).getUsersList()
+      .stream()
+      .filter(u -> !"admin".equals(u.getLogin()))
+      .forEach(
+        user -> users.deactivate(user.getLogin()));
+  }
+
+  @Test
+  public void authenticate_simple_user() {
+    initSamlSettings();
+    UserRepresentation userRepresentation = keycloakServices.createUser();
```

Comment:
I'm worried if there is multiple test in parallel this will produce false positive. This can happen if there is 2 PR (or someone executing a test from its computer) on sonar-auth-saml. I would suggest to prefix all users to have a login starting with a static random string on this test or to create users with random login and keeping a list of those ones to be deleted after the test.